### ISSUE TITLE
cleaner solution for tools.jar dependency

### ISSUE THREAD

**Issue by fommil** [awhitford/lombok.maven#10]
This is really a stub to link with an RFE filed with the Maven compiler plugin for a cleaner way to depend on tools.jar:

  http://jira.codehaus.org/browse/MCOMPILER-175

**Comment by awhitford** [comment]
Sounds like this will all change with Java 9 and [JEP 220: Modular Run-Time Images](http://openjdk.java.net/jeps/220).

**Comment by fommil** [comment]
oh, several years later, I've had to solve this problem for scala projects. This is how we do it in the sbt build tool, I'm sure something similar could be done in Maven https://github.com/ensime/ensime-server/blob/master/project/EnsimeBuild.scala#L216

**Comment by awhitford** [comment]
I did a little more research on this...  Identifying the `tools.jar` is not a problem.  The main problem that I have is **adding** the `tools.jar` to the *Class Realm* of the plugin.  Note that `project.getClassRealm()` returns *null*.  (I tried just adding `tools.jar` to the *class path* for *Delombok*, but that did not do the trick.)

If someone can point me to a solution as to how one might go about adding a jar file to the maven plugin's class loader -- from inside the plugin implementation (not by adding stuff to the pom), then perhaps this can be addressed.
